### PR TITLE
Add toggle group for different types of resources

### PR DESCRIPTION
**Pull request by chenz4027**
Signed-off-by: Maggie Chen <magchen@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx [hunk 1]
```diff
@@ -69,69 +153,160 @@ export default function AdvancedConfiguration() {
...
+                    plural={t('Applications')}
+                    columns={selectedResources.columns}
+                    keyFn={keyFn}
+                    items={selectedResources.items}
+                    extraToolbarControls={
+                        <QuerySwitcher
+                            key="switcher"
+                            options={options.map(({ id }) => ({
+                                id,
+                                contents: t(`application:resource.${id}`),
```
Comment (chenz4027) [inline frontend/src/routes/Applications/AdvancedConfiguration.tsx:276]: We would need the id to determine which toogle is selected and render the corresponding title. Thus I would need useTranslation('application'). @RobDolares would this affect your translation work that you commit in https://github.com/open-cluster-management/console/pull/986 ?

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx [hunk 2]
```diff
@@ -12,41 +17,120 @@ import {
...
 import { useTranslation } from 'react-i18next'
 import { useRecoilState } from 'recoil'
-import { applicationsState } from '../../atoms'
+import { channelsState, placementRulesState, placementsState, subscriptionsState } from '../../atoms'
 import { IResource } from '../../resources'
+import _ from 'lodash'
 
 export default function AdvancedConfiguration() {
-    const { t } = useTranslation()
+    const { t } = useTranslation('application')
```
Comment (chenz4027) [inline frontend/src/routes/Applications/AdvancedConfiguration.tsx:35]: Not sure if this application would affect any translation work.

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx (hunk 1 above)
Comment (chenz4027) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:276]: If so, would you recommend any alternative ways of finding the title?

Comment (KevinFCormier) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:276]: Can you add the title to your metadata above and use `selectedResources.title`?

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx (hunk 2 above)
Comment (KevinFCormier) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:35]: I think the idea is to have a single file with translations (ie. not specify 'applications' here). The keys are just the english strings themselves. So then later, there is a tool to run to collect these into a single file. @RobDolares @jamestalton please correct me if I'm wrong.

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx (hunk 1 above)
Comment (chenz4027) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:276]: Sure that's def doable. Thanks Kevin 👍

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx (hunk 2 above)
Comment (RobDolares) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:35]: @KevinFCormier @chenz4027 - That is correct. we won't reference a specific translation file.  Instead the full text needs to be in the file where it will be translated.  We are still not sure how dynamic string injection is going to be handled, but in cases where enums are used for a dynamic string, i have been creating `switch` statements to return all possible values until we can determine best way to handle string interpolation. @jamestalton

Comment (chenz4027) [reply frontend/src/routes/Applications/AdvancedConfiguration.tsx:35]: Thanks Robert, I'll make the change accordingly.

File: frontend/public/locales/en/application.json [hunk 3]
```diff
@@ -57,6 +57,10 @@
     "repo.type.label.chart": "Chart name",
     "repo.type.label.packageFilterVersion": "Package version",
     "repo.type.label.bucketPath": "Subfolder",
+    "resource.channels": "Channels",
+    "resource.placementrules": "Placement rules",
+    "resource.placements": "Placements",
+    "resource.subscriptions": "Subscriptions",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/application.json:63]: We can revert this change, right? And if you can confirm if this file is no longer used, let's remove it.

File: frontend/src/resources/placement.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        apiVersion: PlacementApiVersion,
+        kind: PlacementKind,
+    })
+    return {
+        promise: placements.promise.then((placements) => {
+            return placements.map((placement) => placement.metadata.name).filter(isType)
+        }),
+        abort: placements.abort,
+    }
+}
```
Comment (KevinFCormier) [inline frontend/src/resources/placement.ts:55]: Do we need this function? Could we just use the atom and useRecoilState hook?

File: frontend/src/routes/Applications/AdvancedConfiguration.tsx [hunk 5]
```diff
@@ -69,69 +234,173 @@ export default function AdvancedConfiguration() {
...
+            { id: 'placementrules', title: 'Placement rules' },
+        ]
+
+        const selectedId = getSelectedId({ location, options, defaultOption, queryParam: 'resources' })
+        const selectedResources = _.get(table, `${selectedId}`)
+
+        return (
+            <AcmTablePaginationContextProvider localStorageKey="advanced-tables-pagination">
+                <AcmTable<IResource>
+                    plural={t('Applications')}
```
Comment (KevinFCormier) [inline frontend/src/routes/Applications/AdvancedConfiguration.tsx:361]: Use the correct title here

File: frontend/src/routes/Applications/CreateApplication/CreateApplication.tsx [hunk 6]
```diff
@@ -64,19 +64,39 @@ export default function CreateApplicationPage() {
 export function CreateApplication() {
     // will need to pass argoNs to AppForm to get argo namespaces
     const [argoNs, setArgoNs] = useState<string[]>([])
+    const [availablePlacements, setAvailablePlacements] = useState<string[]>([])
+
     useEffect(() => {
         const fetchNs = async () => {
             try {
                 let newNs = await listAvailableArgoServerNS().promise
+                let newPlacements = await listPlacements().promise
```
Comment (KevinFCormier) [inline frontend/src/routes/Applications/CreateApplication/CreateApplication.tsx:73]: Can we get both of these from Recoil state instead?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]

File: frontend/public/locales/en/application.json (hunk 3 above)
Comment (chenz4027) [reply frontend/public/locales/en/application.json]: Seems like overview is still using it.

File: frontend/src/resources/resource.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ export interface IResourceDefinition {
 }
 
 export interface IResource extends IResourceDefinition {
+    spec?: any
```
Comment (KevinFCormier) [inline frontend/src/resources/resource.ts:13]: Please remove this and instead set the type parameter correctly on line 150 of console/frontend/src/atoms.tsx

File: frontend/src/resources/gitops-cluster.ts [hunk 8]
```diff
@@ -33,13 +32,3 @@ export function listGitOpsClusters() {
         kind: GitOpsClusterKind,
     })
 }
```
Comment (KevinFCormier) [inline frontend/src/resources/gitops-cluster.ts:34]: Looks like listGitOpsClusters is also unused now?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]

**Comment by chenz4027** [comment]
@KevinFCormier Removed the unused function and refactored the atom instead of the IResource :)
